ui/components/app/snaps/snap-ui-renderer/components/footer.ts

```diff
@@ -77,9 +77,10 @@ export const footer: UIComponentFactory<FooterElement> = ({
       props: {
         ...buttonMapped.props,
         variant:
-          providedChildren.length === 2 && index === 0
+          buttonMapped.props?.variant ??
```

Comment:
@GuillaumeRx Should take a look at this, I'm assuming we weren't intending to let users override this in all cases

Reply:
Nope it was just here to inject the "default" buttons. We only support primary buttons to be defined, the secondary is for the default buttons we add in the renderer

Reply:
This logic should be reverted

Reply:
Should we pass in `snapVariant` (or similar) and just use that for the `danger` check? I'm assuming we are ignoring all other variant info in the footer

Reply:
Oh nevermind, this should be good. If it's a snap action it's always primary.

Reply:
This doesn't break

Reply:
> Should we pass in `snapVariant` (or similar) and just use that for the `danger` check? I'm assuming we are ignoring all other variant info in the footer

Yes I think it would be better, we only use secondary for the default buttons we inject

Reply:
@david0xd Can we try reverting the changes `variant` in the component and add `snapVariant` as a prop that we use to check for `danger`?